packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -455,6 +455,21 @@ class IrisGridTableModelTemplate<
     return !this.isSaveInProgress && this.inputTable != null;
   }
 
+  get hasPendingOperations(): boolean {
+    if (this.viewport == null || this.viewportData == null) {
+      return true;
+    }
+
+    // offset + row.length is last row of loaded data
+    // offset is first row of loaded data
+    // last row of loaded data < last row of viewport
+    // first row of viewport < first row of loaded data
+    return (
+      this.viewportData.offset + this.viewportData.rows.length <
+        this.viewport.bottom || this.viewport.top < this.viewportData.offset
+    );
+  }
```

Comment:
We should also check that the columns requested match the columns in the viewport for horizontal scrolling with a large table. 

A table with 100 columns: 
```
from deephaven import empty_table
import time

def f(a):
    time.sleep(0.1)
    return a

t = empty_table(100).update([f"X{c} = i" for c in range(100)]).update_view("Y = f(i)")
```